File: packages/shared/src/domain/notification.ts
/**
 * Notification Domain Model
 * 
 * Represents a notification in the system.
 */

export type NotificationType = 
  | 'debate_starting' 
  | 'debate_reminder' 
  | 'follow' 
  | 'reaction' 
  | 'comment' 
  | 'mention'
  | 'general';

export interface Notification {
  id: string;
  userId: string;
  type: NotificationType;
  title: string;
  message: string;
  debateId?: string | null;
  debateTitle?: string | null;
  postId?: string | null;
  actorId?: string | null;
  actorName?: string | null;
  actorHandle?: string | null;
  read: boolean;
  createdAt: Date | string;
}

export interface CreateNotificationRequest {
  userId: string;
  type: NotificationType;
  title: string;
  message: string;
  debateId?: string;
  debateTitle?: string;
  postId?: string;
  actorId?: string;
}

export interface NotificationListParams {
  limit?: number;
  offset?: number;
}

export interface UnreadCountResponse {
  count: number;
}

File: backend/internal/models/notification.go
package models

import "time"

type Notification struct {
	ID          string    `json:"id"`
	UserID      string    `json:"userId"`
	Type        string    `json:"type"` // debate_starting, debate_reminder, follow, comment, reaction, mention
	Title       string    `json:"title"`
	Message     string    `json:"message"`
	DebateID    *string   `json:"debateId,omitempty"`
	DebateTitle *string   `json:"debateTitle,omitempty"`
	PostID      *string   `json:"postId,omitempty"`
	ActorID     *string   `json:"actorId,omitempty"`     // User who triggered the notification
	ActorName   *string   `json:"actorName,omitempty"`
	ActorHandle *string   `json:"actorHandle,omitempty"`
	Read        bool      `json:"read"`
	CreatedAt   time.Time `json:"createdAt"`
}

type CreateNotificationRequest struct {
	UserID      string  `json:"userId" validate:"required"`
	Type        string  `json:"type" validate:"required"`
	Title       string  `json:"title" validate:"required"`
	Message     string  `json:"message" validate:"required"`
	DebateID    *string `json:"debateId,omitempty"`
	DebateTitle *string `json:"debateTitle,omitempty"`
	PostID      *string `json:"postId,omitempty"`
	ActorID     *string `json:"actorId,omitempty"`
}

